src/briefcase/commands/dev.py

```diff
@@ -18,6 +18,25 @@ class DevCommand(RunAppMixin, BaseCommand):
     output_format = None
     description = "Run a Briefcase project in the dev environment."
 
+    # On macOS CoreFoundation/NSApplication will do its own independent parsing of argc/argv.
+    # This means that whatever we pass to the Python interpreter on start-up will also be
+    # visible to NSApplication which will interpret things like `-u` (used to make I/O
+    # unbuffered in CPython) as `-u [URL]` (a request to open a document by URL). This is,
+    # rather patently, Not Good.
+    # To avoid this causing unwanted hilarity, we use environment variables to configure the
+    # Python interpreter rather than command-line options.
+    DEV_ENVIRONMENT = {
+        # Equivalent of passing "-u"
+        "PYTHONUNBUFFERED": "1",
+        # Equivalent of passing "-X dev"
+        "PYTHONMALLOC": "debug",
+        "PYTHONASYNCIODEBUG": "1",
+        "PYTHONFAULTHANDLER": "1",
+        "PYTHONWARNINGS": "default",
```

Comment:
Curious....was there a reason that `PYTHONDEVMODE=1` wasn't used instead?

Setting just [PYTHONDEVMODE](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONDEVMODE) seems more likely to be forward-compatible with enabling this mode.

Reply:
...we didn't realise the setting existed :-)

It's obvious now that I'm looking at the docs; no objection replacing these 4 settings with the single PYTHONDEVMODE=1